File: file/validator.go
package file

//go:generate counterfeiter -o ./fakeValidator.go --fake-name fakeValidator ./ Validator

import (
	"errors"
	"fmt"
	"github.com/golang-interfaces/ios"
)

type Validator interface {
	// Validate validates a file
	Validate(
		value *string,
	) []error
}

func newValidator() Validator {
	return _validator{
		os: ios.New(),
	}
}

type _validator struct {
	os ios.IOS
}

// validateFile validates an value against a file parameter
func (this _validator) Validate(
	value *string,
) []error {
	if nil == value {
		return []error{errors.New("file required")}
	}

	fileInfo, err := this.os.Stat(*value)
	if nil != err {
		return []error{err}
	} else if !fileInfo.Mode().IsRegular() {
		return []error{fmt.Errorf("%v not a file", *value)}
	}
	return []error{}
}

File: lib/op/interpreter/opcall/params/param/boolean/validator.js
class Validator {
  /**
   * validates value against constraints
   * @param {boolean} value
   * @return {Array<Error>}
   */
  validate (value) {
    if (typeof value !== 'boolean') {
      return [new Error('boolean required')]
    }

    return []
  }
}

// export as singleton
module.exports = new Validator()
